backend/routes/partners.py

```diff
@@ -193,3 +203,278 @@ def add_member_to_partner(partner_id: int):
+                          sender=TestingConfig.MAIL_USERNAME,
+                          recipients=[body.email])
+            msg.body = """You are not a registered user of NPDC and were
+                        invited to a partner organization. Please register
+                        with NPDC index at
+                        https://dev.nationalpolicedata.org/."""
+            mail.send(msg)
+
+            return {
+                "status": "ok",
+                "message": """User is not registered with the NPDC index.
+                 Email sent to user notifying them to register."""
+            }, 200
+
+        except Exception:
+            return {
+                "status": "error",
+                "message": "Something went wrong! Please try again!"
+            }, 500
+
+
+# admin can remove any member from a partner organization
+@bp.route("/remove_member", methods=['DELETE'])
+@jwt_required()
+@min_role_required(MemberRole.ADMIN)
+def remove_member():
+    body = request.get_json()
+    try:
+        user_found = PartnerMember.query.filter_by(
+            user_id=body["user_id"],
+            partner_id=body["partner_id"]
+            ).delete()
+        db.session.commit()
+        if user_found > 0:
+            return {
+                "status" : "ok",
+                "message" : "Member successfully deleted from Organization"
+            } , 200
+        else:
+            return {
+                "status" : "Error",
+                "message" : "Member is not part of the Organization"
+
+            } , 400
+    except Exception as e:
+        db.session.rollback()
+        return str(e)
+    finally:
+        db.session.close()
+
+
+# admin can withdraw invitations that have been sent out
+@bp.route("/withdraw_invitation", methods=['DELETE'])
+@jwt_required()
+@min_role_required(MemberRole.ADMIN)
+def withdraw_invitation():
+    body = request.get_json()
+    try:
+        user_found = Invitation.query.filter_by(
+            user_id=body["user_id"],
+            partner_id=body["partner_id"]
+            ).delete()
+        db.session.commit()
+        if user_found > 0:
+            return {
+                "status" : "ok",
+                "message" : "Member's invitation withdrawn from Organization"
+            } , 200
+        else:
+            return {
+                "status" : "Error",
+                "message" : "Member is not invited to the Organization"
+
+            } , 400
+    except Exception as e:
+        db.session.rollback()
+        return str(e)
+    finally:
+        db.session.close()
+
+
+# admin can change roles of any user
+@bp.route("/role_change", methods=["PATCH"])
+@jwt_required()
+@min_role_required(MemberRole.ADMIN)
+def role_change():
+    body = request.get_json()
+    try:
+        user_found = PartnerMember.query.filter_by(
+            user_id=body["user_id"],
+            partner_id=body["partner_id"]
+            ).first()
+        if user_found:
```

Comment:
Let's make sure the user in question isn't also an admin. Admins shouldn't be able to demote another admin.